test/google/gax/api_call/retry/raise_test.rb

```diff
@@ -0,0 +1,149 @@
+# Copyright 2019, Google Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+#     * Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#     * Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# 'AS IS' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+require 'test_helper'
+
+class ApiCallRetryTest < Minitest::Test
+  def test_no_retry_without_codes
+    call_count = 0
+    api_meth_stub = proc do
+      call_count += 1
+      raise GRPC::Unavailable.new
+    end
+
+    api_call = Google::Gax::ApiCall.new(api_meth_stub)
+
+    options = Google::Gax::CallOptions.new # no codes
+    assert_raises Google::Gax::GaxError do
+      api_call.call Object.new, options: options
+    end
+    assert_equal(1, call_count)
+  end
+
+  def test_no_retry_with_mismatched_grpc_error
+    api_meth_stub = proc do
+      raise GRPC::Unimplemented.new
+    end
+    api_call = Google::Gax::ApiCall.new(api_meth_stub)
+
+    options = Google::Gax::CallOptions.new(
+      retry_policy: { retry_codes: [GRPC::Core::StatusCodes::UNAVAILABLE] }
+    )
+    assert_raises Google::Gax::GaxError do
+      api_call.call Object.new, options: options
+    end
+  end
+
+  def test_no_retry_with_fake_grpc_error
```

Comment:
It might be useful to add a similar test here that adds a custom policy that does retry this error.

Reply:
Added.